app/lib/pre_assembly/digital_object.rb

```diff
@@ -134,15 +134,17 @@ def build_structural
       end
     end
 
-    def build_from_staging_location(objects:, content_metadata_creation:, content_md_creation_style:, reading_order:)
-      all_paths = objects.flatten.map do |obj|
+    def common_path
+      all_paths = object_files.sort.flatten.map do |obj|
         raise "File '#{obj.path}' not found" unless obj.file_exists?
 
         obj.path # collect all of the filenames into an array
       end
 
-      common_path = Assembly::ObjectFile.common_path(all_paths) # find common paths to all files provided
+      Assembly::ObjectFile.common_path(all_paths) # find common paths to all files provided
+    end
 
```

Comment:
this is a small refactor to make `common_path` a method, so we can use the computed value above and in this existing `build_from_staging_location` method (which has not changed and also makes use of `common_path`)